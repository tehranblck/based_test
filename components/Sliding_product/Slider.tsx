import React from "react";
import Image from "next/image";

export type SliderItem = {
    src: string;
    alt: string;
};

const items: SliderItem[] = [
    { src: "/slide_1.webp", alt: "brand-1" },
    { src: "/slide_2.png", alt: "brand-2" },
    { src: "/slide_1.webp", alt: "brand-3" },
    { src: "/slide_1.webp", alt: "brand-4" },
    { src: "/slide_1.webp", alt: "brand-5" },
    { src: "/slide_1.webp", alt: "brand-6" },
    { src: "/slide_1.webp", alt: "brand-7" },
    { src: "/slide_1.webp", alt: "brand-8" },
];

export default function Slider({ data = items }: { data?: SliderItem[] }) {
    const fadeMask = {
        WebkitMaskImage: "linear-gradient(90deg, transparent, black 10%, black 90%, transparent)",
        maskImage: "linear-gradient(90deg, transparent, black 10%, black 90%, transparent)",
    } as React.CSSProperties;

    const Track = ({ duplicate, className }: { duplicate?: boolean, className?: string }) => (
        <div className="flex items-center gap-10 sm:gap-14 md:gap-16 whitespace-nowrap will-change-transform animate-[marquee-rtl_28s_linear_infinite]" aria-hidden={duplicate}>
            {data.map((item, idx) => (
                <div key={(duplicate ? "dup-" : "") + idx} className={`relative scale-75 h-8 sm:h-10 md:h-12 w-[110px] sm:w-[140px] md:w-[160px] opacity-80 transition-opacity hover:opacity-100 ${className}`}>
                    <Image src={item.src} alt={item.alt} fill className="object-contain" sizes="(max-width: 640px) 110px, (max-width: 768px) 140px, 160px" />
                </div>
            ))}
        </div>
    );

    return (
        <div className="w-full py-6 sm:py-8">
            <div className="relative w-full overflow-hidden" style={fadeMask}>
                <div className="flex w-max">
                    <Track className="mr-16" />
                    <Track duplicate />
                </div>
            </div>
        </div>
    );
}
