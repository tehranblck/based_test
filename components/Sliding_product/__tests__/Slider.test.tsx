import React from 'react'
import { render, screen } from '@testing-library/react'
import Slider from '@/components/Sliding_product/Slider'

describe('Slider', () => {
    it('renders images', () => {
        render(<Slider data={[{ src: '/slide_1.webp', alt: 'brand-1' }, { src: '/slide_2.png', alt: 'brand-2' }]} />)
        expect(screen.getByRole('img', { name: 'brand-1' })).toBeInTheDocument()
        expect(screen.getByRole('img', { name: 'brand-2' })).toBeInTheDocument()
    })

    it('matches snapshot', () => {
        const { container } = render(<Slider />)
        expect(container).toMatchSnapshot()
    })
})


