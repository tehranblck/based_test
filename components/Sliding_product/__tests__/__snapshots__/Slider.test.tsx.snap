// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Slider matches snapshot 1`] = `
<div>
  <div
    class="w-full py-6 sm:py-8"
  >
    <div
      class="relative w-full overflow-hidden"
      style="mask-image: linear-gradient(90deg, transparent, black 10%, black 90%, transparent);"
    >
      <div
        class="flex w-max"
      >
        <div
          class="flex items-center gap-10 sm:gap-14 md:gap-16 whitespace-nowrap will-change-transform animate-[marquee-rtl_28s_linear_infinite]"
        >
          <div
            class="relative scale-75 h-8 sm:h-10 md:h-12 w-[110px] sm:w-[140px] md:w-[160px] opacity-80 transition-opacity hover:opacity-100 mr-16"
          >
            <img
              alt="brand-1"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="(max-width: 640px) 110px, (max-width: 768px) 140px, 160px"
              src="/_next/image?url=%2Fslide_1.webp&w=3840&q=75"
              srcset="/_next/image?url=%2Fslide_1.webp&w=16&q=75 16w, /_next/image?url=%2Fslide_1.webp&w=32&q=75 32w, /_next/image?url=%2Fslide_1.webp&w=48&q=75 48w, /_next/image?url=%2Fslide_1.webp&w=64&q=75 64w, /_next/image?url=%2Fslide_1.webp&w=96&q=75 96w, /_next/image?url=%2Fslide_1.webp&w=128&q=75 128w, /_next/image?url=%2Fslide_1.webp&w=256&q=75 256w, /_next/image?url=%2Fslide_1.webp&w=384&q=75 384w, /_next/image?url=%2Fslide_1.webp&w=640&q=75 640w, /_next/image?url=%2Fslide_1.webp&w=750&q=75 750w, /_next/image?url=%2Fslide_1.webp&w=828&q=75 828w, /_next/image?url=%2Fslide_1.webp&w=1080&q=75 1080w, /_next/image?url=%2Fslide_1.webp&w=1200&q=75 1200w, /_next/image?url=%2Fslide_1.webp&w=1920&q=75 1920w, /_next/image?url=%2Fslide_1.webp&w=2048&q=75 2048w, /_next/image?url=%2Fslide_1.webp&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="relative scale-75 h-8 sm:h-10 md:h-12 w-[110px] sm:w-[140px] md:w-[160px] opacity-80 transition-opacity hover:opacity-100 mr-16"
          >
            <img
              alt="brand-2"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="(max-width: 640px) 110px, (max-width: 768px) 140px, 160px"
              src="/_next/image?url=%2Fslide_2.png&w=3840&q=75"
              srcset="/_next/image?url=%2Fslide_2.png&w=16&q=75 16w, /_next/image?url=%2Fslide_2.png&w=32&q=75 32w, /_next/image?url=%2Fslide_2.png&w=48&q=75 48w, /_next/image?url=%2Fslide_2.png&w=64&q=75 64w, /_next/image?url=%2Fslide_2.png&w=96&q=75 96w, /_next/image?url=%2Fslide_2.png&w=128&q=75 128w, /_next/image?url=%2Fslide_2.png&w=256&q=75 256w, /_next/image?url=%2Fslide_2.png&w=384&q=75 384w, /_next/image?url=%2Fslide_2.png&w=640&q=75 640w, /_next/image?url=%2Fslide_2.png&w=750&q=75 750w, /_next/image?url=%2Fslide_2.png&w=828&q=75 828w, /_next/image?url=%2Fslide_2.png&w=1080&q=75 1080w, /_next/image?url=%2Fslide_2.png&w=1200&q=75 1200w, /_next/image?url=%2Fslide_2.png&w=1920&q=75 1920w, /_next/image?url=%2Fslide_2.png&w=2048&q=75 2048w, /_next/image?url=%2Fslide_2.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="relative scale-75 h-8 sm:h-10 md:h-12 w-[110px] sm:w-[140px] md:w-[160px] opacity-80 transition-opacity hover:opacity-100 mr-16"
          >
            <img
              alt="brand-3"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="(max-width: 640px) 110px, (max-width: 768px) 140px, 160px"
              src="/_next/image?url=%2Fslide_1.webp&w=3840&q=75"
              srcset="/_next/image?url=%2Fslide_1.webp&w=16&q=75 16w, /_next/image?url=%2Fslide_1.webp&w=32&q=75 32w, /_next/image?url=%2Fslide_1.webp&w=48&q=75 48w, /_next/image?url=%2Fslide_1.webp&w=64&q=75 64w, /_next/image?url=%2Fslide_1.webp&w=96&q=75 96w, /_next/image?url=%2Fslide_1.webp&w=128&q=75 128w, /_next/image?url=%2Fslide_1.webp&w=256&q=75 256w, /_next/image?url=%2Fslide_1.webp&w=384&q=75 384w, /_next/image?url=%2Fslide_1.webp&w=640&q=75 640w, /_next/image?url=%2Fslide_1.webp&w=750&q=75 750w, /_next/image?url=%2Fslide_1.webp&w=828&q=75 828w, /_next/image?url=%2Fslide_1.webp&w=1080&q=75 1080w, /_next/image?url=%2Fslide_1.webp&w=1200&q=75 1200w, /_next/image?url=%2Fslide_1.webp&w=1920&q=75 1920w, /_next/image?url=%2Fslide_1.webp&w=2048&q=75 2048w, /_next/image?url=%2Fslide_1.webp&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="relative scale-75 h-8 sm:h-10 md:h-12 w-[110px] sm:w-[140px] md:w-[160px] opacity-80 transition-opacity hover:opacity-100 mr-16"
          >
            <img
              alt="brand-4"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="(max-width: 640px) 110px, (max-width: 768px) 140px, 160px"
              src="/_next/image?url=%2Fslide_1.webp&w=3840&q=75"
              srcset="/_next/image?url=%2Fslide_1.webp&w=16&q=75 16w, /_next/image?url=%2Fslide_1.webp&w=32&q=75 32w, /_next/image?url=%2Fslide_1.webp&w=48&q=75 48w, /_next/image?url=%2Fslide_1.webp&w=64&q=75 64w, /_next/image?url=%2Fslide_1.webp&w=96&q=75 96w, /_next/image?url=%2Fslide_1.webp&w=128&q=75 128w, /_next/image?url=%2Fslide_1.webp&w=256&q=75 256w, /_next/image?url=%2Fslide_1.webp&w=384&q=75 384w, /_next/image?url=%2Fslide_1.webp&w=640&q=75 640w, /_next/image?url=%2Fslide_1.webp&w=750&q=75 750w, /_next/image?url=%2Fslide_1.webp&w=828&q=75 828w, /_next/image?url=%2Fslide_1.webp&w=1080&q=75 1080w, /_next/image?url=%2Fslide_1.webp&w=1200&q=75 1200w, /_next/image?url=%2Fslide_1.webp&w=1920&q=75 1920w, /_next/image?url=%2Fslide_1.webp&w=2048&q=75 2048w, /_next/image?url=%2Fslide_1.webp&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="relative scale-75 h-8 sm:h-10 md:h-12 w-[110px] sm:w-[140px] md:w-[160px] opacity-80 transition-opacity hover:opacity-100 mr-16"
          >
            <img
              alt="brand-5"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="(max-width: 640px) 110px, (max-width: 768px) 140px, 160px"
              src="/_next/image?url=%2Fslide_1.webp&w=3840&q=75"
              srcset="/_next/image?url=%2Fslide_1.webp&w=16&q=75 16w, /_next/image?url=%2Fslide_1.webp&w=32&q=75 32w, /_next/image?url=%2Fslide_1.webp&w=48&q=75 48w, /_next/image?url=%2Fslide_1.webp&w=64&q=75 64w, /_next/image?url=%2Fslide_1.webp&w=96&q=75 96w, /_next/image?url=%2Fslide_1.webp&w=128&q=75 128w, /_next/image?url=%2Fslide_1.webp&w=256&q=75 256w, /_next/image?url=%2Fslide_1.webp&w=384&q=75 384w, /_next/image?url=%2Fslide_1.webp&w=640&q=75 640w, /_next/image?url=%2Fslide_1.webp&w=750&q=75 750w, /_next/image?url=%2Fslide_1.webp&w=828&q=75 828w, /_next/image?url=%2Fslide_1.webp&w=1080&q=75 1080w, /_next/image?url=%2Fslide_1.webp&w=1200&q=75 1200w, /_next/image?url=%2Fslide_1.webp&w=1920&q=75 1920w, /_next/image?url=%2Fslide_1.webp&w=2048&q=75 2048w, /_next/image?url=%2Fslide_1.webp&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="relative scale-75 h-8 sm:h-10 md:h-12 w-[110px] sm:w-[140px] md:w-[160px] opacity-80 transition-opacity hover:opacity-100 mr-16"
          >
            <img
              alt="brand-6"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="(max-width: 640px) 110px, (max-width: 768px) 140px, 160px"
              src="/_next/image?url=%2Fslide_1.webp&w=3840&q=75"
              srcset="/_next/image?url=%2Fslide_1.webp&w=16&q=75 16w, /_next/image?url=%2Fslide_1.webp&w=32&q=75 32w, /_next/image?url=%2Fslide_1.webp&w=48&q=75 48w, /_next/image?url=%2Fslide_1.webp&w=64&q=75 64w, /_next/image?url=%2Fslide_1.webp&w=96&q=75 96w, /_next/image?url=%2Fslide_1.webp&w=128&q=75 128w, /_next/image?url=%2Fslide_1.webp&w=256&q=75 256w, /_next/image?url=%2Fslide_1.webp&w=384&q=75 384w, /_next/image?url=%2Fslide_1.webp&w=640&q=75 640w, /_next/image?url=%2Fslide_1.webp&w=750&q=75 750w, /_next/image?url=%2Fslide_1.webp&w=828&q=75 828w, /_next/image?url=%2Fslide_1.webp&w=1080&q=75 1080w, /_next/image?url=%2Fslide_1.webp&w=1200&q=75 1200w, /_next/image?url=%2Fslide_1.webp&w=1920&q=75 1920w, /_next/image?url=%2Fslide_1.webp&w=2048&q=75 2048w, /_next/image?url=%2Fslide_1.webp&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="relative scale-75 h-8 sm:h-10 md:h-12 w-[110px] sm:w-[140px] md:w-[160px] opacity-80 transition-opacity hover:opacity-100 mr-16"
          >
            <img
              alt="brand-7"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="(max-width: 640px) 110px, (max-width: 768px) 140px, 160px"
              src="/_next/image?url=%2Fslide_1.webp&w=3840&q=75"
              srcset="/_next/image?url=%2Fslide_1.webp&w=16&q=75 16w, /_next/image?url=%2Fslide_1.webp&w=32&q=75 32w, /_next/image?url=%2Fslide_1.webp&w=48&q=75 48w, /_next/image?url=%2Fslide_1.webp&w=64&q=75 64w, /_next/image?url=%2Fslide_1.webp&w=96&q=75 96w, /_next/image?url=%2Fslide_1.webp&w=128&q=75 128w, /_next/image?url=%2Fslide_1.webp&w=256&q=75 256w, /_next/image?url=%2Fslide_1.webp&w=384&q=75 384w, /_next/image?url=%2Fslide_1.webp&w=640&q=75 640w, /_next/image?url=%2Fslide_1.webp&w=750&q=75 750w, /_next/image?url=%2Fslide_1.webp&w=828&q=75 828w, /_next/image?url=%2Fslide_1.webp&w=1080&q=75 1080w, /_next/image?url=%2Fslide_1.webp&w=1200&q=75 1200w, /_next/image?url=%2Fslide_1.webp&w=1920&q=75 1920w, /_next/image?url=%2Fslide_1.webp&w=2048&q=75 2048w, /_next/image?url=%2Fslide_1.webp&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="relative scale-75 h-8 sm:h-10 md:h-12 w-[110px] sm:w-[140px] md:w-[160px] opacity-80 transition-opacity hover:opacity-100 mr-16"
          >
            <img
              alt="brand-8"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="(max-width: 640px) 110px, (max-width: 768px) 140px, 160px"
              src="/_next/image?url=%2Fslide_1.webp&w=3840&q=75"
              srcset="/_next/image?url=%2Fslide_1.webp&w=16&q=75 16w, /_next/image?url=%2Fslide_1.webp&w=32&q=75 32w, /_next/image?url=%2Fslide_1.webp&w=48&q=75 48w, /_next/image?url=%2Fslide_1.webp&w=64&q=75 64w, /_next/image?url=%2Fslide_1.webp&w=96&q=75 96w, /_next/image?url=%2Fslide_1.webp&w=128&q=75 128w, /_next/image?url=%2Fslide_1.webp&w=256&q=75 256w, /_next/image?url=%2Fslide_1.webp&w=384&q=75 384w, /_next/image?url=%2Fslide_1.webp&w=640&q=75 640w, /_next/image?url=%2Fslide_1.webp&w=750&q=75 750w, /_next/image?url=%2Fslide_1.webp&w=828&q=75 828w, /_next/image?url=%2Fslide_1.webp&w=1080&q=75 1080w, /_next/image?url=%2Fslide_1.webp&w=1200&q=75 1200w, /_next/image?url=%2Fslide_1.webp&w=1920&q=75 1920w, /_next/image?url=%2Fslide_1.webp&w=2048&q=75 2048w, /_next/image?url=%2Fslide_1.webp&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </div>
        <div
          aria-hidden="true"
          class="flex items-center gap-10 sm:gap-14 md:gap-16 whitespace-nowrap will-change-transform animate-[marquee-rtl_28s_linear_infinite]"
        >
          <div
            class="relative scale-75 h-8 sm:h-10 md:h-12 w-[110px] sm:w-[140px] md:w-[160px] opacity-80 transition-opacity hover:opacity-100 undefined"
          >
            <img
              alt="brand-1"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="(max-width: 640px) 110px, (max-width: 768px) 140px, 160px"
              src="/_next/image?url=%2Fslide_1.webp&w=3840&q=75"
              srcset="/_next/image?url=%2Fslide_1.webp&w=16&q=75 16w, /_next/image?url=%2Fslide_1.webp&w=32&q=75 32w, /_next/image?url=%2Fslide_1.webp&w=48&q=75 48w, /_next/image?url=%2Fslide_1.webp&w=64&q=75 64w, /_next/image?url=%2Fslide_1.webp&w=96&q=75 96w, /_next/image?url=%2Fslide_1.webp&w=128&q=75 128w, /_next/image?url=%2Fslide_1.webp&w=256&q=75 256w, /_next/image?url=%2Fslide_1.webp&w=384&q=75 384w, /_next/image?url=%2Fslide_1.webp&w=640&q=75 640w, /_next/image?url=%2Fslide_1.webp&w=750&q=75 750w, /_next/image?url=%2Fslide_1.webp&w=828&q=75 828w, /_next/image?url=%2Fslide_1.webp&w=1080&q=75 1080w, /_next/image?url=%2Fslide_1.webp&w=1200&q=75 1200w, /_next/image?url=%2Fslide_1.webp&w=1920&q=75 1920w, /_next/image?url=%2Fslide_1.webp&w=2048&q=75 2048w, /_next/image?url=%2Fslide_1.webp&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="relative scale-75 h-8 sm:h-10 md:h-12 w-[110px] sm:w-[140px] md:w-[160px] opacity-80 transition-opacity hover:opacity-100 undefined"
          >
            <img
              alt="brand-2"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="(max-width: 640px) 110px, (max-width: 768px) 140px, 160px"
              src="/_next/image?url=%2Fslide_2.png&w=3840&q=75"
              srcset="/_next/image?url=%2Fslide_2.png&w=16&q=75 16w, /_next/image?url=%2Fslide_2.png&w=32&q=75 32w, /_next/image?url=%2Fslide_2.png&w=48&q=75 48w, /_next/image?url=%2Fslide_2.png&w=64&q=75 64w, /_next/image?url=%2Fslide_2.png&w=96&q=75 96w, /_next/image?url=%2Fslide_2.png&w=128&q=75 128w, /_next/image?url=%2Fslide_2.png&w=256&q=75 256w, /_next/image?url=%2Fslide_2.png&w=384&q=75 384w, /_next/image?url=%2Fslide_2.png&w=640&q=75 640w, /_next/image?url=%2Fslide_2.png&w=750&q=75 750w, /_next/image?url=%2Fslide_2.png&w=828&q=75 828w, /_next/image?url=%2Fslide_2.png&w=1080&q=75 1080w, /_next/image?url=%2Fslide_2.png&w=1200&q=75 1200w, /_next/image?url=%2Fslide_2.png&w=1920&q=75 1920w, /_next/image?url=%2Fslide_2.png&w=2048&q=75 2048w, /_next/image?url=%2Fslide_2.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="relative scale-75 h-8 sm:h-10 md:h-12 w-[110px] sm:w-[140px] md:w-[160px] opacity-80 transition-opacity hover:opacity-100 undefined"
          >
            <img
              alt="brand-3"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="(max-width: 640px) 110px, (max-width: 768px) 140px, 160px"
              src="/_next/image?url=%2Fslide_1.webp&w=3840&q=75"
              srcset="/_next/image?url=%2Fslide_1.webp&w=16&q=75 16w, /_next/image?url=%2Fslide_1.webp&w=32&q=75 32w, /_next/image?url=%2Fslide_1.webp&w=48&q=75 48w, /_next/image?url=%2Fslide_1.webp&w=64&q=75 64w, /_next/image?url=%2Fslide_1.webp&w=96&q=75 96w, /_next/image?url=%2Fslide_1.webp&w=128&q=75 128w, /_next/image?url=%2Fslide_1.webp&w=256&q=75 256w, /_next/image?url=%2Fslide_1.webp&w=384&q=75 384w, /_next/image?url=%2Fslide_1.webp&w=640&q=75 640w, /_next/image?url=%2Fslide_1.webp&w=750&q=75 750w, /_next/image?url=%2Fslide_1.webp&w=828&q=75 828w, /_next/image?url=%2Fslide_1.webp&w=1080&q=75 1080w, /_next/image?url=%2Fslide_1.webp&w=1200&q=75 1200w, /_next/image?url=%2Fslide_1.webp&w=1920&q=75 1920w, /_next/image?url=%2Fslide_1.webp&w=2048&q=75 2048w, /_next/image?url=%2Fslide_1.webp&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="relative scale-75 h-8 sm:h-10 md:h-12 w-[110px] sm:w-[140px] md:w-[160px] opacity-80 transition-opacity hover:opacity-100 undefined"
          >
            <img
              alt="brand-4"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="(max-width: 640px) 110px, (max-width: 768px) 140px, 160px"
              src="/_next/image?url=%2Fslide_1.webp&w=3840&q=75"
              srcset="/_next/image?url=%2Fslide_1.webp&w=16&q=75 16w, /_next/image?url=%2Fslide_1.webp&w=32&q=75 32w, /_next/image?url=%2Fslide_1.webp&w=48&q=75 48w, /_next/image?url=%2Fslide_1.webp&w=64&q=75 64w, /_next/image?url=%2Fslide_1.webp&w=96&q=75 96w, /_next/image?url=%2Fslide_1.webp&w=128&q=75 128w, /_next/image?url=%2Fslide_1.webp&w=256&q=75 256w, /_next/image?url=%2Fslide_1.webp&w=384&q=75 384w, /_next/image?url=%2Fslide_1.webp&w=640&q=75 640w, /_next/image?url=%2Fslide_1.webp&w=750&q=75 750w, /_next/image?url=%2Fslide_1.webp&w=828&q=75 828w, /_next/image?url=%2Fslide_1.webp&w=1080&q=75 1080w, /_next/image?url=%2Fslide_1.webp&w=1200&q=75 1200w, /_next/image?url=%2Fslide_1.webp&w=1920&q=75 1920w, /_next/image?url=%2Fslide_1.webp&w=2048&q=75 2048w, /_next/image?url=%2Fslide_1.webp&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="relative scale-75 h-8 sm:h-10 md:h-12 w-[110px] sm:w-[140px] md:w-[160px] opacity-80 transition-opacity hover:opacity-100 undefined"
          >
            <img
              alt="brand-5"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="(max-width: 640px) 110px, (max-width: 768px) 140px, 160px"
              src="/_next/image?url=%2Fslide_1.webp&w=3840&q=75"
              srcset="/_next/image?url=%2Fslide_1.webp&w=16&q=75 16w, /_next/image?url=%2Fslide_1.webp&w=32&q=75 32w, /_next/image?url=%2Fslide_1.webp&w=48&q=75 48w, /_next/image?url=%2Fslide_1.webp&w=64&q=75 64w, /_next/image?url=%2Fslide_1.webp&w=96&q=75 96w, /_next/image?url=%2Fslide_1.webp&w=128&q=75 128w, /_next/image?url=%2Fslide_1.webp&w=256&q=75 256w, /_next/image?url=%2Fslide_1.webp&w=384&q=75 384w, /_next/image?url=%2Fslide_1.webp&w=640&q=75 640w, /_next/image?url=%2Fslide_1.webp&w=750&q=75 750w, /_next/image?url=%2Fslide_1.webp&w=828&q=75 828w, /_next/image?url=%2Fslide_1.webp&w=1080&q=75 1080w, /_next/image?url=%2Fslide_1.webp&w=1200&q=75 1200w, /_next/image?url=%2Fslide_1.webp&w=1920&q=75 1920w, /_next/image?url=%2Fslide_1.webp&w=2048&q=75 2048w, /_next/image?url=%2Fslide_1.webp&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="relative scale-75 h-8 sm:h-10 md:h-12 w-[110px] sm:w-[140px] md:w-[160px] opacity-80 transition-opacity hover:opacity-100 undefined"
          >
            <img
              alt="brand-6"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="(max-width: 640px) 110px, (max-width: 768px) 140px, 160px"
              src="/_next/image?url=%2Fslide_1.webp&w=3840&q=75"
              srcset="/_next/image?url=%2Fslide_1.webp&w=16&q=75 16w, /_next/image?url=%2Fslide_1.webp&w=32&q=75 32w, /_next/image?url=%2Fslide_1.webp&w=48&q=75 48w, /_next/image?url=%2Fslide_1.webp&w=64&q=75 64w, /_next/image?url=%2Fslide_1.webp&w=96&q=75 96w, /_next/image?url=%2Fslide_1.webp&w=128&q=75 128w, /_next/image?url=%2Fslide_1.webp&w=256&q=75 256w, /_next/image?url=%2Fslide_1.webp&w=384&q=75 384w, /_next/image?url=%2Fslide_1.webp&w=640&q=75 640w, /_next/image?url=%2Fslide_1.webp&w=750&q=75 750w, /_next/image?url=%2Fslide_1.webp&w=828&q=75 828w, /_next/image?url=%2Fslide_1.webp&w=1080&q=75 1080w, /_next/image?url=%2Fslide_1.webp&w=1200&q=75 1200w, /_next/image?url=%2Fslide_1.webp&w=1920&q=75 1920w, /_next/image?url=%2Fslide_1.webp&w=2048&q=75 2048w, /_next/image?url=%2Fslide_1.webp&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="relative scale-75 h-8 sm:h-10 md:h-12 w-[110px] sm:w-[140px] md:w-[160px] opacity-80 transition-opacity hover:opacity-100 undefined"
          >
            <img
              alt="brand-7"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="(max-width: 640px) 110px, (max-width: 768px) 140px, 160px"
              src="/_next/image?url=%2Fslide_1.webp&w=3840&q=75"
              srcset="/_next/image?url=%2Fslide_1.webp&w=16&q=75 16w, /_next/image?url=%2Fslide_1.webp&w=32&q=75 32w, /_next/image?url=%2Fslide_1.webp&w=48&q=75 48w, /_next/image?url=%2Fslide_1.webp&w=64&q=75 64w, /_next/image?url=%2Fslide_1.webp&w=96&q=75 96w, /_next/image?url=%2Fslide_1.webp&w=128&q=75 128w, /_next/image?url=%2Fslide_1.webp&w=256&q=75 256w, /_next/image?url=%2Fslide_1.webp&w=384&q=75 384w, /_next/image?url=%2Fslide_1.webp&w=640&q=75 640w, /_next/image?url=%2Fslide_1.webp&w=750&q=75 750w, /_next/image?url=%2Fslide_1.webp&w=828&q=75 828w, /_next/image?url=%2Fslide_1.webp&w=1080&q=75 1080w, /_next/image?url=%2Fslide_1.webp&w=1200&q=75 1200w, /_next/image?url=%2Fslide_1.webp&w=1920&q=75 1920w, /_next/image?url=%2Fslide_1.webp&w=2048&q=75 2048w, /_next/image?url=%2Fslide_1.webp&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="relative scale-75 h-8 sm:h-10 md:h-12 w-[110px] sm:w-[140px] md:w-[160px] opacity-80 transition-opacity hover:opacity-100 undefined"
          >
            <img
              alt="brand-8"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="(max-width: 640px) 110px, (max-width: 768px) 140px, 160px"
              src="/_next/image?url=%2Fslide_1.webp&w=3840&q=75"
              srcset="/_next/image?url=%2Fslide_1.webp&w=16&q=75 16w, /_next/image?url=%2Fslide_1.webp&w=32&q=75 32w, /_next/image?url=%2Fslide_1.webp&w=48&q=75 48w, /_next/image?url=%2Fslide_1.webp&w=64&q=75 64w, /_next/image?url=%2Fslide_1.webp&w=96&q=75 96w, /_next/image?url=%2Fslide_1.webp&w=128&q=75 128w, /_next/image?url=%2Fslide_1.webp&w=256&q=75 256w, /_next/image?url=%2Fslide_1.webp&w=384&q=75 384w, /_next/image?url=%2Fslide_1.webp&w=640&q=75 640w, /_next/image?url=%2Fslide_1.webp&w=750&q=75 750w, /_next/image?url=%2Fslide_1.webp&w=828&q=75 828w, /_next/image?url=%2Fslide_1.webp&w=1080&q=75 1080w, /_next/image?url=%2Fslide_1.webp&w=1200&q=75 1200w, /_next/image?url=%2Fslide_1.webp&w=1920&q=75 1920w, /_next/image?url=%2Fslide_1.webp&w=2048&q=75 2048w, /_next/image?url=%2Fslide_1.webp&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
